metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -479,9 +656,249 @@ def emit_chart_mces(self) -> Iterable[MetadataWorkUnit]:
+                    and cll_info.downstream.column is not None
+                    else []
+                )
+                upstreams = [
+                    make_schema_field_urn(column_ref.table, column_ref.column)
+                    for column_ref in cll_info.upstreams
+                ]
+                fine_grained_lineages.append(
+                    FineGrainedLineageClass(
+                        downstreamType=FineGrainedLineageDownstreamTypeClass.FIELD,
+                        downstreams=downstream,
+                        upstreamType=FineGrainedLineageUpstreamTypeClass.FIELD_SET,
+                        upstreams=upstreams,
+                    )
+                )
+
+            upstream_lineage = UpstreamLineageClass(
+                upstreams=[
+                    UpstreamClass(
+                        type=DatasetLineageTypeClass.TRANSFORMED,
+                        dataset=input_table_urn,
+                    )
+                    for input_table_urn in parsed_query_object.in_tables
+                ],
+                fineGrainedLineages=fine_grained_lineages,
+            )
+        else:
+            ## To account for Physical datasets
+            tag_urn = f"urn:li:tag:{self.platform}:physical"
+            upstream_dataset = self.get_datasource_urn_from_id(dataset_response, db_platform_instance)
+            upstream_lineage = UpstreamLineageClass(
+                upstreams=[
+                    UpstreamClass(
+                        type=DatasetLineageTypeClass.TRANSFORMED,
+                        dataset=upstream_dataset,
+                    )
+                ]
+            )
+        global_tags = GlobalTagsClass(tags=[TagAssociationClass(tag=tag_urn)])
+
+        dataset_snapshot = DatasetSnapshot(
+            urn=datasource_urn,
+            aspects=[self.gen_schema_metadata(datasource_urn, dataset_response), dataset_info, glossary_terms, upstream_lineage, global_tags],
+        )
+        return dataset_snapshot
+
+    def emit_dataset_mces(self) -> Iterable[MetadataWorkUnit]:
+        current_dataset_page = 0
+        # We will set total datasets to the actual number after we get the response
+        total_datasets = PAGE_SIZE
+
+        while current_dataset_page * PAGE_SIZE <= total_datasets:
+            full_dataset_response = self.session.get(
+                f"{self.config.connect_uri}/api/v1/dataset/",
+                params=f"q=(page:{current_dataset_page},page_size:{PAGE_SIZE})",
+            )
+            if full_dataset_response.status_code != 200:
+                logger.warning(f"Failed to get dataset data: {full_dataset_response.text}")
+            full_dataset_response.raise_for_status()
+
+            current_dataset_page += 1
+
+            payload = full_dataset_response.json()
+            total_datasets = payload["count"]
+            for dataset_data in payload["result"]:
+                if dataset_data.get("id") != 287:
```

Comment:
Not sure whether this is some (old) testing code? Or why should a specific dataset be skipped here?

Reply:
Yes, this was for testing specific dataset. Will look to clean up PR/Write tests